### PR TITLE
Add Key Features Introduction and update existed doc

### PR DESCRIPTION
**Pull request by JesseStutler**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What this PR does / why we need it**:
The official website documentation needs to be improved, and a `Features Introduction` section needs to be added to introduce features and how to use them. This PR does such following things:
1. Improved title/code block style
2. Add code block copy button
3. Fix some js problem
4. Add Key Features item and following doc:
- Hierarchical Queue
- Queue Resource Management
- Unified Scheduling
5. Besides, Reconstructed queue.md, moved the usage scenario to queue resource management, and then updated some fields

* **Which issue(s) this PR fixes**:
related issue: #330

Check deploy/netlify to see the details

### CODE REVIEW

File: config/_default/menus.toml [hunk 1]
```diff
@@ -73,23 +73,28 @@
   identifier = "concepts"
 
 [[docs]]
-  name = "Ecosystem"
+  name = "Features Introduction"
```
Comment (Monokaix) [inline config/_default/menus.toml:76]: "Key Features" is better.

File: config/_default/languages.toml [hunk 2]
```diff
@@ -97,23 +97,28 @@
   identifier = "concepts"
 
 [[zh.menu.docs]]
-  name = "生态"
+  name = "特性介绍"
```
Comment (Monokaix) [inline config/_default/languages.toml:100]: 关键特性

File: content/zh/docs/hierarchical_queue.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+type = "docs"  # Do not modify.
+
+# Add menu entry to sidebar.
+[menu.docs]
+  parent = "features"
+  weight = 1
++++
+
+## 背景
+在多租户场景中，队列是实现公平调度、资源隔离和任务优先级控制的核心机制。然而，在目前的Volcano中，队列仅支持平级结构，缺乏层级概念。而在实际应用中，不同队列往往隶属于不同部门，部门之间存在层级关系，对资源的分配和抢占需求也更为精细。为此，Volcano v2.0引入了层级队列功能，大幅增强了队列的能力。通过这一功能，用户可以在层级队列的基础上实现更细粒度的资源配额管理和抢占策略，构建更高效的统一调度平台。
```
Comment (Monokaix) [inline content/zh/docs/hierarchical_queue.md:18]: 2.0 is not released yet, we can change it to "Volcano latest版本".

Comment (JesseStutler) [reply content/zh/docs/hierarchical_queue.md]: updated

File: config/_default/languages.toml (hunk 2 above)
Comment (JesseStutler) [reply config/_default/languages.toml]: updated

File: config/_default/menus.toml (hunk 1 above)
Comment (JesseStutler) [reply config/_default/menus.toml]: updated

File: content/en/docs/queue_based_resource_management.md
Comment (hwdef) [inline content/en/docs/queue_based_resource_management.md:1]: This file is empty.

Comment (JesseStutler) [reply content/en/docs/queue_based_resource_management.md:1]: I'm still working on adding more docs into this PR, I may add WIP to hold this PR

File: content/en/docs/queue_resource_management.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+   * Resource preemption: Ensure resource requirements for high-priority tasks
+
+4. Multi-tenant Isolation
+
+   * Strict resource quota control
+   * Priority-based resource allocation
+   * Prevent single tenant from excessive resource usage
+
+## Queue Scheduling Implementation
+### Related Scheduling Actions
```
Comment (Monokaix) [inline content/en/docs/queue_resource_management.md:58]: We should emphasize 
it's queue related actions.

File: content/en/docs/queue_resource_management.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+Queue scheduling in Volcano involves the following core Actions:
+
+1. Enqueue Action: Controls job admission into queues, determines whether to allow new jobs based on queue resource quotas and current usage.
```
Comment (Monokaix) [inline content/en/docs/queue_resource_management.md:61]: Add the notice that enqueue action conflict with preempt/reclaim.

File: content/en/docs/hierarchical_queue.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    - plugins:
+      - name: priority
+      - name: gang
+        enablePreemptable: false
+    - plugins:
+      - name: drf
+        enablePreemptable: false
+      - name: predicates
+      - name: capacity
+        enableHierarchy: true
```
Comment (Monokaix) [inline content/en/docs/hierarchical_queue.md:50]: We'd better also add comments here about what should be configured.

File: content/zh/docs/queue_resource_management.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+   * 此时资源分配比例为 default:test = 1:3
+4. 在test队列创建job3，申请3C资源
+   * 结果：系统回收default队列超出deserved的资源
+   * job2（3C）被驱逐
+   * job1（1C）保留运行
+   * job3（3C）开始运行
+
+这个场景同时适用于Capacity Plugin和Proportion Plugin：
+
+   * Capacity Plugin：直接配置deserved值（default=1C, test=3C）
```
Comment (Monokaix) [inline content/zh/docs/queue_resource_management.md:138]: Plugin name should be lower case to match the plugin name itself and we can also link to the plugin in scheduler.plugins part.

File: content/zh/docs/queue_resource_management.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+      > 建议: 
+      > 1. 进行三级资源配置时，需遵循: Guarantee <= Deserved <= Capability；
+      > 2. Guarantee可按需配置，在开启Capacity插件时需要配置Deserved值；
+      > 3. Deserved配置建议：在平级队列场景，所有队列的Deserved值总和等于集群资源总量；在层级队列场景，子队列的Deserved值总和等于父队列的Deserved值。
+
+   * 支持动态资源配额调整
+
+
+2.[层级队列](/zh/docs/hierarchical_queue)管理
```
Comment (Monokaix) [inline content/zh/docs/queue_resource_management.md:39]: 1 and 2 are not aligned.

File: content/zh/docs/queue_resource_management.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+spec:
+  weight: 1
+  capability:
+    cpu: "20"
+    memory: "40Gi"
+```
+Proportion插件可根据资源总量和队列数量自动调整队列的Deserved值，配置简单直观，适用于多租场景下各租户权重相同，对资源精确度要求不高的场景。如集群总资源为`total_resource`,
+此时有多个队列中同时提交了作业，则其中一个队列设置的Deserved值即为`(weight/total_weight)*total_resource`
+
+#### 使用场景示例
```
Comment (Monokaix) [inline content/zh/docs/queue_resource_management.md:116]: `使用样例` is enough.

File: content/zh/docs/queue_resource_management.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+1. Enqueue Action：控制作业进入队列的准入机制，根据队列的资源配额和当前使用情况决定是否允许新作业进入队列。
+
+2. Allocate Action：负责资源分配过程，确保分配符合队列配额限制，同时支持队列间的资源借用机制，提高资源利用率。
+
+3. Preempt Action：支持**队列内**资源抢占。高优先级作业可以抢占同队列内低优先级作业的资源，确保关键任务的及时执行。
+
+4. Reclaim Action：支持**队列间**的资源回收。当队列资源紧张时，触发资源回收机制。优先回收超出队列deserved值的资源，并结合队列/作业优先级选择合适的牺牲者。
+
+### 队列调度插件
+Volcano提供了两个核心的队列调度插件，它们都是基于Deserved值进行资源管理，区别在于Deserved值的设置方式：：
```
Comment (Monokaix) [inline content/zh/docs/queue_resource_management.md:71]: `deserved` lower case.

File: content/zh/docs/queue_resource_management.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+   * 当队列资源使用超过Deserved值时，超出部分可被其他队列回收
+   * 当其他队列资源空闲时，队列可以临时使用超过Deserved值的资源
+
+2. 资源回收机制
+   * 触发条件：集群资源紧张且某个队列需要使用资源时
+   * 回收对象：使用超过Deserved值的队列
+   * 回收过程：逐步回收超出部分，直到符合Deserved值
+
+#### Capacity Plugin
+Capacity Plugin支持通过显式配置Deserved值来设置队列资源应得量，如以下队列配置示例：
```
Comment (Monokaix) [inline content/zh/docs/queue_resource_management.md:84]: We should emphasize that capacity plugin is a static way to setting deserved value and users may need to change the deserved spec when clusters resources changed, while proportion is a dynamic way and users no need to change it.

File: content/zh/docs/unified_scheduling.md [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+
+- 强大的批处理调度能力：通过 VcJob 完美支持 TensorFlow、PyTorch、Ray、Spark 等主流 AI 和大数据框架
+- 完整的 Kubernetes 工作负载支持：可直接调度 Deployment、StatefulSet、Job、DaemonSet 等原生工作负载
+
+这种统一调度能力让用户能够使用单一调度器来管理所有类型的工作负载，极大简化了集群管理复杂度。
+
+## 兼容Kubernetes调度能力
+
+### 1. 节点过滤机制 (Predicates)
+Volcano完整实现了Kube-Scheduler中的核心节点过滤机制,包括：
```
Comment (Monokaix) [inline content/zh/docs/unified_scheduling.md:26]: We should emphasize that the k8s related features are hosted by predict and nodeorder seperately.
And remove (Predicates) and (NodeOrder).

File: content/zh/docs/unified_scheduling.md [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+   - 适用于 100+ 节点的集群
+   - 可显著减少调度延迟
+
+> **注意**：
+>
+> - 只有静态的检查结果才会被缓存（如节点标签、污点等）
+> - 动态资源相关的检查（如 CPU、内存用量）每次都会重新计算
+> - 当节点状态发生变化时，相关缓存会自动失效
+
+#### 资源比例预留 (ProportionalPredicate)
```
Comment (Monokaix) [inline content/zh/docs/unified_scheduling.md:79]: This part is not a key feature and we can remove it to `Plugins` part.

File: content/zh/docs/queue_resource_management.md [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+
+#### 使用场景示例
+
+##### 资源回收场景
+###### 背景：
+* 集群CPU总量为4C
+* 已创建default队列（weight=1或deserved=1C）
+* 集群中无任务运行
+
+###### 操作步骤：
```
Comment (Monokaix) [inline content/zh/docs/queue_resource_management.md:124]: Add the whole yamls is better.

File: content/zh/docs/queue_resource_management.md [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+  name: proportion-queue
+spec:
+  weight: 1
+  capability:
+    cpu: "20"
+    memory: "40Gi"
+```
+Proportion插件可根据资源总量和队列数量自动调整队列的Deserved值，配置简单直观，适用于多租场景下各租户权重相同，对资源精确度要求不高的场景。如集群总资源为`total_resource`,
+此时有多个队列中同时提交了作业，则其中一个队列设置的Deserved值即为`(weight/total_weight)*total_resource`
+
```
Comment (Monokaix) [inline content/zh/docs/queue_resource_management.md:115]: Add the notice that capacity and proportion are two different queue management way and cannot use together.

File: content/zh/docs/unified_scheduling.md [hunk 16]
```diff
@@ -0,0 +1,263 @@
...
+[menu.docs]
+  parent = "features"
+  weight = 3
++++
+
+## 概述
+
+Volcano 作为业界领先的云原生批处理系统调度器，通过统一的调度系统实现了对所有工作负载类型的支持：
+
+- 强大的批处理调度能力：通过 VcJob 完美支持 TensorFlow、PyTorch、Ray、Spark 等主流 AI 和大数据框架
```
Comment (Monokaix) [inline content/zh/docs/unified_scheduling.md:18]: Ray、TensorFlow、PyTorch、MindSPore、Spark、Flink

File: content/en/docs/queue_resource_management.md (hunk 4 above)
Comment (JesseStutler) [reply content/en/docs/queue_resource_management.md:58]: image
updated

File: content/en/docs/queue_resource_management.md (hunk 5 above)
Comment (JesseStutler) [reply content/en/docs/queue_resource_management.md:61]: image
updated

File: content/en/docs/hierarchical_queue.md (hunk 6 above)
Comment (JesseStutler) [reply content/en/docs/hierarchical_queue.md:50]: image
updated

File: content/zh/docs/queue_resource_management.md (hunk 7 above)
Comment (JesseStutler) [reply content/zh/docs/queue_resource_management.md:237]: updated

File: content/zh/docs/queue_resource_management.md (hunk 8 above)
Comment (JesseStutler) [reply content/zh/docs/queue_resource_management.md]: updated

image

File: content/zh/docs/queue_resource_management.md (hunk 9 above)
Comment (JesseStutler) [reply content/zh/docs/queue_resource_management.md]: updated

File: content/zh/docs/queue_resource_management.md (hunk 10 above)
Comment (JesseStutler) [reply content/zh/docs/queue_resource_management.md]: updated

File: content/zh/docs/queue_resource_management.md (hunk 11 above)
Comment (JesseStutler) [reply content/zh/docs/queue_resource_management.md]: image

updated

File: content/zh/docs/unified_scheduling.md (hunk 12 above)
Comment (JesseStutler) [reply content/zh/docs/unified_scheduling.md]: updated, now use predicates plugin and nodeorder plugin

File: content/zh/docs/unified_scheduling.md (hunk 13 above)
Comment (JesseStutler) [reply content/zh/docs/unified_scheduling.md]: deleted

File: content/zh/docs/queue_resource_management.md (hunk 14 above)
Comment (JesseStutler) [reply content/zh/docs/queue_resource_management.md]: Added

File: content/zh/docs/queue_resource_management.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (JesseStutler) [reply content/zh/docs/queue_resource_management.md]: Already added

File: content/zh/docs/unified_scheduling.md (hunk 16 above)
Comment (JesseStutler) [reply content/zh/docs/unified_scheduling.md]: updated

File: content/zh/docs/queue_resource_management.md [hunk 17]
```diff
@@ -0,0 +1,241 @@
...
+  weight = 1
++++
+
+## 功能概述
+
+[队列](/zh/docs/queue)是Volcano的核心概念之一，用于支持多租户场景下的资源分配与任务调度。通过队列，用户可以实现多租资源分配、任务优先级控制、资源抢占与回收等功能，显著提升集群的资源利用率和任务调度效率。
+
+## 核心特性
+
+### 1. 灵活的资源配置
```
Comment (Monokaix) [inline content/zh/docs/queue_resource_management.md:24]: 缩进有问题

Comment (JesseStutler) [reply content/zh/docs/queue_resource_management.md:24]: solved

**Comment by william-wang** [comment]
@JesseStutler The PR is cool. With these introcution, it's more easier for user to use the feature.
